packages/helpers/src/hooks/useDeterminateState.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * useDeterminateState
+ *
+ * A hook that returns a state transitioning to a determinate value after a minimum duration.
+ * For example, you could use this hook with a button that transitions into a "loading" state,
+ * ensuring it remains in the "loading" state for at least 1000ms. This provides clear feedback
+ * to the user that the action is being processed.
+ *
+ */
+import React from "react";
+
+type UseDeterminateStateParams<T> = {
+  value: T;
+  determinateValue: T;
+  minDurationMs?: number;
+};
+
+const useDeterminateState = <T>({
+  value,
+  determinateValue,
+  minDurationMs = 1000,
+}: UseDeterminateStateParams<T>): T => {
```

Comment:
One future thought is if we should be using type doc or JS doc for all our hooks in telegraph? Not sure on current state of docs tooling so good one to explore

Reply:
Yeah will have to look into that more, would be cool to get descriptions showing up in intellisense